main.go

```diff
@@ -140,6 +136,8 @@ func main() {
 				os.Exit(1)
 			}
 
+			clientCache.AddClient(cCfg.GetContextName(), c.GetClient())
```

Comment:
With this change, I think we could turn the method `ClientCache.CreateClient()` private, or better yet, inline it.